src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.removeFromAlive();
+    }
+  }
+
+  removeFromAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
+
+  bite(damage) {
+    this.health -= damage;
+    this.checkHealth();
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+      herbivore.checkHealth();
+    }
```

Comment:
[CODE PERFORMANCE] It's preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method after decreasing the health of the herbivore. This can be done inside this `bite` method to ensure that the animal is removed from the `Animal.alive` array if its health drops to 0 or below as a result of the bite.